File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    let map= new Map();
    participant.forEach((e) =>{
        map.set(e,  map.has(e) ? map.get(e) + 1 : 1);
    });
    
    completion.forEach((e) =>{
        map.set(e , map.get(e) - 1);
    });
    
    for(let [key , value] of map){
        if(value === 1) return key;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    
    d = dict()
    
    for p in participant:
        if p in d:
            d[p] += 1
        else:
            d[p] = 1
    
    for c in completion:
        if c in d:
            d[c] -= 1
            
    for key, value in d.items():
        if value == 1:
            return key
            